src/live.rs

```diff
@@ -160,6 +165,197 @@ pub fn pxe_ignition_unwrap(config: &PxeIgnitionUnwrapConfig) -> Result<()> {
+        // magic number
+        file.read_exact(&mut buf)
+            .chain_err(|| "reading karg embed header")?;
+        if buf != COREOS_KARG_EMBED_AREA_HEADER_MAGIC {
+            bail!("No karg embed areas found; old or corrupted CoreOS ISO image.");
+        }
+        // length
+        file.read_exact(&mut buf)
+            .chain_err(|| "reading karg embed header")?;
+        let length: usize = u64::from_le_bytes(buf)
+            .try_into()
+            .chain_err(|| "karg embed area length too large to allocate")?;
+        // sanity-check against a reasonable limit
+        if length > COREOS_KARG_EMBED_AREA_MAX_SIZE {
+            bail!(
+                "karg embed area length larger than {} (found {})",
+                COREOS_KARG_EMBED_AREA_MAX_SIZE,
+                length
+            );
+        }
+
+        let metadata = file.metadata().chain_err(|| "reading metadata for ISO")?;
+        let iso_size = metadata.len();
+
+        // default kargs
+        file.read_exact(&mut buf)
+            .chain_err(|| "reading karg embed header")?;
+        let default_kargs_offset: u64 = u64::from_le_bytes(buf);
+        if default_kargs_offset > iso_size {
+            bail!(
+                "Offset to default kargs outside ISO ({} vs {})",
+                default_kargs_offset,
+                iso_size
+            );
+        }
+
+        // offsets
+        let mut kargs_offsets: Vec<u64> = Vec::new();
+        while kargs_offsets.len() < COREOS_KARG_EMBED_AREA_HEADER_MAX_OFFSETS {
+            file.read_exact(&mut buf)
+                .chain_err(|| "reading karg embed header")?;
+            let offset: u64 = u64::from_le_bytes(buf);
+            if offset == 0 {
+                break;
+            } else if offset > iso_size {
+                bail!(
+                    "Offset to kargs area outside ISO ({} vs {})",
+                    offset,
+                    iso_size
+                );
+            }
+            kargs_offsets.push(offset);
+        }
+
+        // we expect at least one
+        if kargs_offsets.is_empty() {
+            bail!("No karg embed areas found; corrupted CoreOS ISO image.");
+        }
+
+        Ok(KargEmbedAreas {
+            file,
+            length,
+            default_kargs_offset,
+            kargs_offsets,
+        })
+    }
+
+    fn get_current_kargs(&mut self) -> Result<String> {
+        // really, we could just get the kargs from the first file, but let's sanity-check that all
+        // the offsets have the same kargs
+        let mut first_kargs: Option<String> = None;
+        for offset in &self.kargs_offsets {
+            let kargs = get_kargs_at_offset(self.file, self.length, *offset)?;
+            if let Some(ref first_kargs) = first_kargs {
+                if &kargs != first_kargs {
+                    bail!(
+                        "kargs don't match at all offsets! (defaults are '{}', but offset {} has: '{}')",
```

Comment:
Not actually the default kargs.  `expected '{}'`?

Reply:
Ahh yup, good catch.